client/src/components/QA/QA.jsx

```diff
@@ -20,32 +21,40 @@ class QA extends React.Component {
     };
   }
 
-  componentDidMount() {
-    axios.get(`/qa/questions?product_id=${this.state.productId}&count=1000`)
-      .then(res => {
-        this.setState({
-          questions: res.data.results
-        });
-      })
-      .catch(err => {
-        console.log(err);
+  fetchData() {
+    const { updateCache, checkCache, selectedProduct } = this.props;
+    const questionsCache = checkCache('questions', selectedProduct.id);
+    if (questionsCache) {
+      this.setState({
+        questions: questionsCache,
+        productId: selectedProduct.id
       });
-  }
-
-  componentDidUpdate() {
-    if (this.state.productId !== this.props.selectedProduct.id) {
-      axios.get(`/qa/questions?product_id=${this.props.selectedProduct.id}&count=1000`)
+    } else {
+      axios.get(`/qa/questions?product_id=${selectedProduct.id}&count=1000`)
         .then(res => {
+          updateCache('questions', selectedProduct.id, res.data.results);
           this.setState({
             questions: res.data.results,
-            productId: this.props.selectedProduct.id
+            productId: selectedProduct.id
           });
         })
         .catch(err => {
           console.log(err);
         });
     }
   }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate() {
+    const {selectedProduct} = this.props;
+    if (this.state.productId !== selectedProduct.id) {
+      this.fetchData();
+    }
+  }
+
   updateData() {
```

Comment:
Is `updateData` unused?

Reply:
`updateData`  I use it for I update the new data after I post question.